doc/developer-guides/timer-hld.rst

```diff
@@ -0,0 +1,98 @@
+.. _timer-hld:
+
+Timer Management high-level design
+###################################
+
+This document describes timer management for the ACRN hypervisor.
+
+Overview
+********
+
+Timer management support is limited to only use LAPIC tsc-deadline timer
+as the clock source. Also, you can only add a timer on the logical CPU
+for the process or thread.  Timer scheduling and timer migrating are
+forbidden.
+
+How it works
+************
+
+When the system boots, we check that the hardware supports a LAPIC
+tsc-deadline timer by checking ``CPUID.01H:ECX.TSC_Deadline[bit 24]``.
+If support is missing, we output an error message and panic the
+hypervisor.  Otherwise, we register the timer interrupt callback that
+raises a timer software interrupt on each logical CPU and set the lapic
+timer mode to tsc-deadline timer mode by writing the locap APIC LVT
+register.
+
+timer.h API
+***********
+
+The timer data structures are defined in
+``hypervisor/include/arch/x86/timer.h``:
+
+.. code-block:: c
+
+   struct per_cpu_timers {
+      struct list_head timer_list;    /* active timer list */
+   };
+
+   struct hv_timer {
+      struct list_head node;      /* link-list for all timers */
+      int mode;                   /* timer mode: one-shot or periodic */
+      uint64_t fire_tsc;          /* tsc deadline to interrupt */
+      uint64_t period_in_cycle;   /* period of the periodic timer in units of TSC cycles */
+      timer_handle_t func;        /* callback when time reached */
+      void *priv_data;            /* callback function private data */
+   };
```

Comment:
Some of the explanations above are a little different (more verbose) than what is in the header file. The ones here seem better, should we align the ones in the code to match the HLD?